alacritty/Cargo.toml

```diff
@@ -42,6 +42,7 @@ image = { version = "0.23.3", default-features = false, features = ["ico"] }
 
 [target.'cfg(target_os = "macos")'.dependencies]
 objc = "0.2.2"
+darwin-libproc = "0.2"
```

Comment:
My biggest concern with this PR are the two new dependencies pulled in here. Unlike libc, these are basically unused and seem to be passively maintained by just a single person. So I think if we could circumvent these it would be beneficial.

Reply:
@chrisduerr I'm extremely conflicted about this personally...

Before I responded I had a bit of a walk through of the code inside `darwin-libproc`, and found that they aren't building the binding code in a build script, but instead using a fixed run of `bindgen`. So either way I've opened heim-rs/darwin-libproc#2. So I guess I'll see how that goes...

Assuming there's not a horrible issue with that process, I feel like working with the community here is better than writing our own logic, even if it's for a very small application. I personally don't want to maintain this logic very much, and I expect the users of `darwin-libproc` and `heim` in general will spot issues in addition to ourselves. There does seem to be some macOS version dependence, but I'm not sure why we'd want to handle that ourselves any differently than anyone else. I'm also not sure if there's any versions of macOS we care about without the features we need.

Having read over the implementation of [`pid_info`](https://github.com/heim-rs/darwin-libproc/blob/dcc1a4ba7d22164cf7e1656a64e814440bbe4742/darwin-libproc/src/pid_info.rs#L4-L30) now, I'm not sure if there's anything I would do differently. I couldn't find any documentation from Apple for `libproc` past the header itself really, so I just have to go on sane assumptions and the testing I did locally on my one Mac laptop I have lying around.

Everything seems to work, so I'm not sure if this should be merged, or if we should avoid this issue altogether. But that's kinda where I'm at right now.

Reply:
> I feel like working with the community here is better than writing our own logic

There is no "community". It's just some random person I do not know or trust at all.

> I expect the users of darwin-libproc and heim in general will spot issues

It seems like that has already failed, with them using `bindgen` improperly.

> I'm also not sure if there's any versions of macOS we care about without the features we need.

That should be very clear and decided upon independent from this issue. Currently at least 10.11+ should be supported.

Reply:
I'd like to wait to see what they say about my PR for at least a couple days. If there's no response, or I'm unsatisfied, we can move on. I'm curious if you'd rather see a hand written binding, or our own use of `bindgen` though?

> That should be very clear and decided upon independent from this issue. Currently at least 10.11+ should be supported.

Seems like the value for the "flavor" argument of this function call hasn't changed since [10.11](https://git.privacyone.io/useful/macos-sdk/-/blob/master/MacOSX10.11.sdk/usr/include/sys/proc_info.h), since it's still 9 on my machine.

Reply:
> I'm curious if you'd rather see a hand written binding, or our own use of bindgen though?

I don't care particularly much about this detail. I feel like for a single function a handwritten one is probably going to be easier?

Reply:
> I feel like for a single function a handwritten one is probably going to be easier?

I don't think so... `bindgen` works **very** well in my experience, and makes all the needed values exposed in a reasonable fashion. Whereas otherwise, I'd need to define a number of constants myself, and ensure they stay consistent with the header.

Reply:
The thing is all we need is the single function, nothing else.

Reply:
Unused code optimizations should make it so we're not bloating our binaries... unless something is wrong with that, I don't see why I should have to maintain these bindings in a more cumbersome way. Fact is, single function or not, `bindgen` is the best way to do this I believe.

Reply:
Pulling in a whole swath of code for just a single function is a bit ridiculous and just unnecessarily complicated. Shoving this work on the code optimizer but having to deal with constantly updating a load of completely unnecessary functionality is not productive.

Reply:
I mean, we already require `bindgen` for `winpty-sys` so I see no harm reusing it here.

Reply:
I suppose technically it's optional. But then again, this is only required on macOS, so it's somewhat similar. Perhaps I need to fix the current state of the `Cargo.lock`?

Reply:
> I mean, we already require bindgen for winpty-sys so I see no harm reusing it here.

You mean the thing that is getting removed in ~10 weeks? My problem is not with a dependency on bindgen, it's just that I don't see how bindgen makes anything easier for us here. It's just a single function definition without a dependency on any other types.

I'm not sure if it might require one or two additional calls for interpreting the `proc_pidinfo` call, but it certainly isn't extensive.

Reply:
I'm mostly talking about the `#define PROC_PIDVNODEPATHINFO		9` definition. Perhaps adding our own constant isn't the end of the world, but I still feel like I'd rather use `bindgen`.

Reply:
A single constant certainly shouldn't be the end of the world.

Reply:
OK, I'll look into making a `macos` specific module in `alacritty` for this definition.

Reply:
Sure. Fyi we already do this for Windows PTY stuff. So it's certainly not the first time.